### PR TITLE
Add core files.

### PR DESCRIPTION
**Pull request by mattcollier**
This is lifted out of bedrock-ledger-node.  Why is the getter asynchronous?  Can we make it synchronous and not have to concern ourselves with callbacks vs promises?

It also occurs to me that this might be a candidate for a BSD licensed thing.

tracking: https://github.com/digitalbazaar/bedrock-ledger-agent/pull/17

### CODE REVIEW

**Comment by davidlehn** [comment]
Yes, the generic injector bits should migrate to their own repo and npm package.  I'd suggest something without the "bedrock-" namespacing.  Similar code exists in BSD licensed packages like did-veres-one, did-client-veres-one, web-request-rpc, and probably others.

**Comment by davidlehn** [comment]
Other than the BedrockError usage, is this independent of bedrock?

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
+*.py[co]
```
Comment (davidlehn) [inline .gitignore]: Probably not going to have python code in this repo.

File: lib/Injector.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    const plugin = self.plugins[capabilityName];
+
+    // asynchronous getter
+    if(callback) {
+      if(!plugin) {
+        return callback(new BedrockError('Plugin not found.', 'NotFoundError', {
+          httpStatusCode: 404,
+          public: true,
```
Comment (davidlehn) [inline lib/Injector.js]: What is the use case for this sort of situation being a public 404?

**Comment by dlongley** [comment]
I think we have slightly different injector patterns in some bedrock libraries vs. the ones @davidlehn mentioned. Not sure how to reconcile -- or if we're actually dealing with two different things that should not be reconciled. Does anyone else have more experience/contact with this problem?

I'm not sure why the getter is asynchronous ... it seems like it was done to accommodate some async (the helper lib) pattern... which I don't think is a good justification for doing it. I would prefer it to be synchronous ... and if it needs to be async in the future the promise pattern can cover that better than the callback one could. I'd like it to be changed to synchronous unless someone can defend it being async.

**Comment by mattcollier** [comment]
@dlongley @davidlehn synchronous now.  Ready for another pass.  If we come up with some BSD thing later that can be plugged in here, great, but this needs to get to a v1 release.

@davidlehn I find you recent obsession with .gitignore files disturbing.

**Comment by mattcollier** [comment]
@dlongley some other merging is being held up by this.  Are we good here?  Are we agreed that we can implement a BSD dependency at a later date?

**Review by dlongley — APPROVED** [review]

**Pull request merged**